uReplicator-Common/src/main/java/com/uber/stream/ureplicator/common/KafkaUReplicatorMetricsReporter.java

```diff
@@ -65,46 +62,36 @@
       _reporterMetricPrefix = null;
       return;
     }
-
-    _reporterMetricPrefix = String
-        .format(KAFKA_UREPLICATOR_METRICS_REPORTER_PREFIX_FORMAT, conf.getRegion(),
-            String.join(".", conf.getAdditionalInfo()),
-            conf.getHostname());
-    LOGGER.info("Reporter Metric Prefix is : " + _reporterMetricPrefix);
     _registry = new MetricRegistry();
-    final Boolean enabledGraphiteReporting = true;
-    final Boolean enabledJmxReporting = true;
-    final long graphiteReportFreqSec = 60L;
 
     // Init jmx reporter
-    if (enabledJmxReporting) {
+    if (conf.getEnabledGraphiteReport()) {
       _jmxReporter = JmxReporter.forRegistry(this._registry).build();
       _jmxReporter.start();
     } else {
       _jmxReporter = null;
     }
 
     // Init graphite reporter
-    if (enabledGraphiteReporting) {
+    if (conf.getEnabledJmxReport()) {
       Graphite graphite = getGraphite(conf);
       if (graphite == null) {
         _graphiteReporter = null;
       } else {
+        _reporterMetricPrefix = String
+                .format(KAFKA_UREPLICATOR_METRICS_REPORTER_PREFIX_FORMAT, conf.getRegion(),
+                        String.join(".", conf.getAdditionalInfo()),
+                        conf.getHostname());
+        LOGGER.info("Reporter Metric Prefix is : {}", _reporterMetricPrefix);
         _graphiteReporter =
             GraphiteReporter.forRegistry(_registry).prefixedWith(_reporterMetricPrefix)
                 .convertRatesTo(TimeUnit.SECONDS)
                 .convertDurationsTo(TimeUnit.MILLISECONDS).filter(MetricFilter.ALL).build(graphite);
-        _graphiteReporter.start(graphiteReportFreqSec, TimeUnit.SECONDS);
+        _graphiteReporter.start(conf.getGraphiteReportFreqSec(), TimeUnit.SECONDS);
       }
     } else {
       _graphiteReporter = null;
     }
-
-    Runtime.getRuntime().addShutdownHook(new Thread() {
```

Comment:
Since we removed this, can we add this to the manager/controller starter?

Reply:
I have added to manager/controller starter already

Reply:
yes, we should do it like this.
worker-3.0 has invoke KafkaUReplicatorMetricsReporter.stop visibly,
but i didn't check it in manager/controller.
now ,  i have added